pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -656,6 +671,9 @@ func (oe *operationExecutor) generateMountVolumeFunc(
 		mountErr := volumeMounter.SetUp(fsGroup)
 		if mountErr != nil {
 			// On failure, return error. Caller will log and retry.
+			if errGetRef == nil && ref != nil {
+				oe.recorder.Eventf(ref, api.EventTypeWarning, kubecontainer.FailedMountVolume, "Unable to mount volumes for pod %q: %v", volumeToMount.PodName, mountErr)
```

Comment:
Let's keep the wording consistent with the returned error:
`MountVolume.SetUp failed for volume %q (spec.Name: %q) pod %q (UID: %q) with: %v`
Maybe just construct the error first, use it to fire an event, then return it.